### ISSUE TITLE
Multithreading "detect" statements causes errors

### ISSUE THREAD

**Issue by gferrell** [emgucv/emgucv#588]
When calling multiple detect statements for MaskRCNN

Windows
.Net Core 5

x86_64

4.5.3.4781
Create list of "detect," tasks, then call them asynchronously so that they overlap.  You will likely see skipped images, errors and have to lock around Init.  Anyway, before posting the whole thing, I would like to ask if EMGU is thread-safe?

**Comment by emgucv** [comment]
No, objects in Open CV, similar to C# List<> class, are not thread safe.

That means, if you create one MaskRCNN instance per thread, you will be fine. But if you create a single MaskRcnn network and call "detect" function (which modified the internals of the network) from multiple threads, you will run into trouble. 

Same principal applies to the C# List<> implementation. If you create one instance per thread you will be fine. If you only perform read access aross several threads, you will be fine. But if you start to modify it across different threads you will run into troubles (racing conditions).

**Comment by emgucv** [comment]
Closing ticket now.
